addons/crm/models/crm_lead.py

```diff
@@ -1197,26 +1197,52 @@ def _get_rainbowman_message(self):
             AND
                 probability = 100
             AND
-                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', CURRENT_DATE)
+                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', %(tz_midnight)s)
             AND
                 (user_id = %(user_id)s OR team_id = %(team_id)s)
         """
-        self.env.cr.execute(query, {'user_id': self.user_id.id,
-                                    'team_id': self.team_id.id})
+        tz_midnight = fields.Datetime.now().astimezone(pytz.timezone(self.user_id.tz or 'UTC')).replace(hour=0, minute=0, second=0)
```

Comment:
Would fallback on user's TZ before probably (not worse than UTC, and in most cases you work in same TZ as your colleagues)
```suggestion
        tz_midnight = fields.Datetime.now().astimezone(pytz.timezone(self.user_id.tz or self.env.user.tz or 'UTC')).replace(hour=0, minute=0, second=0)
```